draft-ietf-lwig-minimal-esp.xml

```diff
@@ -292,52 +292,49 @@ As a result, it should not be part of a minimal implementation of ESP.  </t>
-Note that it is not because an encryption algorithm transform is widely deployed that it is secured. 
-As a result, security should not be weakened for interoperability. 
-<xref target="RFC8221"/> and successors consider the life cycle of encryption algorithm transforms sufficiently long to provide interoperability.  
-Constrained devices may have limited interoperability requirements which makes possible to reduces the number of encryption algorithm transforms to implement.</t>
-
-<t> Power Consumption and Cipher Suite Complexity: Complexity of the encryption algorithm transform or the energy associated with it are especially considered when devices have limited resources or are using some batteries, in which case the battery determines the life of the
-device. 
-The choice of a cryptographic function may consider re-using specific libraries or to take advantage of hardware acceleration provided by the device. 
-For example, if the device benefits from AES hardware modules and uses AES-CTR, it may prefer AUTH_AES-XCBC for its authentication. 
-In addition, some devices may also embed radio modules with hardware acceleration for AES-CCM, in which case, this mode may be preferred.</t>
-
-<t> Power Consumption and Bandwidth Consumption: Similarly to the encryption algorithm transform complexity, reducing the payload sent, may significantly reduce the energy consumption of the device. 
-As a result, encryption algorithm transforms with low overhead may be considered. 
+
+<t> Security: Security is the criteria that should be considered first for the selection of encryption algorithm transform.
+The security of encryption algorithm transforms is expected to evolve over time, and it is of primary importance to follow up-to-date security guidance and recommendations.
+The chosen encryption algorithm must not be vulnerable or weak (see <xref target="RFC8221"/> for outdated ciphers).
+ESP can be used to authenticate only (ENCR_NULL) or to encrypt the communication.
+In the latter case, authenticated encryption (AEAD) is RECOMMENDED <xref target="RFC8221"/>.</t>
+
+<t>Resilience to nonce re-use: Some transforms -including AES-GCM - are vulnerable to nonce collision with a given key.
+While the generation of the nonce may prevent such collision during a session, the mechanisms are unlikely to provide such protection across sleep states or reboot.
+This causes an issue for devices that are configured using static keys (called manual keying) and manual keying should not be used with these encryption algorithms.
+When the key is likely to be re-used across reboots, algorithms that are nonce misuse resistant such as, for example, AES-SIV <xref target="RFC5297"/>, AES-GCM-SIV <xref target="RFC8452"/> or Deoxys-II <xref target="DeoxysII"/> are RECOMMENDED.
+Note however that currently none of these are yet defined for use with ESP.
+[Paul: missing guidance on what to do then for devices using manual keying?]
```

Comment:
I meant you now created a huge problem in this document. Until now, it was assumed some secure keying mechanism, like IKEv2, was used. Since the document states that you should really be using AEADs, and no AEADs can be safely used with manual keying, you would now have to explain what encryption algoritms those devices that only do manual keying need to use. AES-CBC?

Manual keying should never be done. Either keys should be generated from IKE, or keys should be supplied on boot with the presumption that the supplier deals with not issuing the same key twice.

In other words, I think "when a key is likely to be re-used across reboots" is a fundamentally broken and insecure method that no one should ever implement, and no RFC should cater to.

Reply:
A sensor is unlikely to implement IKEv2 and is likely to be provisioned with a key.

Reply:
Then it is also likely provisioned with the SPI for the key/policy - after all whoever provides this has the capability to generate randomness much better.

Reply:
Sure, but in very constraint mode the same SPI MAY be used during the lifetime of the device. The SPI may be randomly generated out of band as well.